docs/brainstorming/consolidation-strategy.md

```diff
@@ -0,0 +1,275 @@
+                      │ DeepIsles        │
+                      │ (External Ref)   │
+                      │                  │
+                      │ • Nature 2025    │
+                      │ • Stroke lesion  │
+                      │   segmentation   │
+                      │ • Docker image   │
+                      └──────────────────┘
+```
+
+## Related Repositories
+
+| Repository | Type | Purpose | Link |
+|------------|------|---------|------|
+| neuroimaging-go-brrrr | Hub | Consolidation hub for neuroimaging tools | [GitHub](https://github.com/CloseChoice/neuroimaging-go-brrrr) |
+| arc-aphasia-bids | Production | Upload ARC dataset (ds004884) to HuggingFace Hub | [GitHub](https://github.com/The-Obstacle-Is-The-Way/arc-aphasia-bids) |
+| stroke-deepisles-demo | Production | DeepISLES inference demo with Gradio UI | [GitHub](https://github.com/The-Obstacle-Is-The-Way/stroke-deepisles-demo) |
+| huggingface/datasets | Consumption | Pending PRs for BIDS loader, lazy upload fixes | [PR #7886](https://github.com/huggingface/datasets/pull/7886) |
+| DeepIsles | Reference | State-of-the-art stroke lesion segmentation | [GitHub](https://github.com/ezequieldlrosa/DeepIsles) |
+
+## HuggingFace Resources
+
+| Resource | Description | Link |
+|----------|-------------|------|
+| arc-aphasia-bids dataset | ARC dataset on HF Hub (293 GB, 902 sessions) | [HF Hub](https://huggingface.co/datasets/hugging-science/arc-aphasia-bids) |
+| bids-neuroimaging Space | NiiVue visualization demo | [HF Spaces](https://huggingface.co/spaces/TobiasPitters/bids-neuroimaging) |
+
+## Key Technical Components
+
+### 1. BIDS Loader (Consumption - PR #7886)
+
+Native BIDS support for HuggingFace's `datasets` library:
+
+```python
+from datasets import load_dataset
+
+# Load any BIDS dataset with one line
+ds = load_dataset('bids', data_dir="path/to/bids/dataset", streaming=True)
+
+# Push to Hub
+ds.push_to_hub("org/dataset-name", num_shards={"train": 500})
+```
+
+Features:
+
+- Uses `pybids` for BIDS-compliant file discovery
+- `Nifti()` feature type for NIfTI file handling (already merged)
+- `Nifti1ImageWrapper._repr_html_()` for NiiVue visualization (already merged)
+- Streaming support for large datasets
+
+### 2. arc-aphasia-bids (Production)
+
+Custom pipeline for the Aphasia Recovery Cohort (ARC) dataset:
+
+- Converts OpenNeuro ds004884 (BIDS) to HuggingFace Dataset format
+- Validation against the Scientific Data paper (230 subjects, 902 sessions)
+- Session-level sharding for memory efficiency
+
+### 3. stroke-deepisles-demo (Production + Inference)
+
+End-to-end inference demo:
+
+- Loads BIDS/HF neuroimaging data
+- Runs DeepISLES Docker container for stroke lesion segmentation
+- Gradio UI for interactive visualization
+- Comprehensive test suite (96 tests, 82% coverage)
+
+## Consolidation Options
+
+### Option A: Modular Monorepo
+
+Migrate all code into `neuroimaging-go-brrrr` as packages:
+
+```text
+neuroimaging-go-brrrr/
```

Comment:
I support this! I think I simply called packages tools, since they are not real python packages as of now. And as it stands, I don't think they need to be